From f915c46102be4b9fbb2ab3a7d8952fdef20c6558 Mon Sep 17 00:00:00 2001
From: Muthukumar K <muthukumar.kandasamy@intel.com>
Date: Sat, 8 Sep 2012 23:25:47 +0530
Subject: Audio stops after seeking AVI clip with MP3 audio (special contents without proper key samples)

BZ: 43872

During seek in AVI, nearest Sync sample was searched both for
audio and video. Sync samples are useful only in case of
video. We can ignore Sync field for non-video tracks.

Category: aosp improvement
Domain: AUDIO_MEDIA-Audio Framework
Origin: Internal
Upstream-Candidate: Yes

Orig-Change-Id: Id406894f8f382ab00800604564d72c3b20329493

Change-Id: I3c5ead2591ade1b42d7ab06f9b4e3ccec0a75a8f
Signed-off-by: Muthukumar K <muthukumar.kandasamy@intel.com>
Reviewed-on: http://android.intel.com:8080/77934
Reviewed-by: Yao, Yong <yong.yao@intel.com>
Reviewed-by: M, Arulselvan <arulselvan.m@intel.com>
Tested-by: M, Arulselvan <arulselvan.m@intel.com>
Reviewed-by: cactus <cactus@intel.com>
Tested-by: cactus <cactus@intel.com>
Signed-off-by: Arulselvan M <arulselvan.m@intel.com>
---
 media/libstagefright/AVIExtractor.cpp |    6 ++++++
 1 files changed, 6 insertions(+), 0 deletions(-)

diff --git a/media/libstagefright/AVIExtractor.cpp b/media/libstagefright/AVIExtractor.cpp
index eb7017f..98069cd 100644
--- a/media/libstagefright/AVIExtractor.cpp
+++ b/media/libstagefright/AVIExtractor.cpp
@@ -1249,6 +1249,12 @@ status_t AVIExtractor::getSampleIndexAtTime(
         return OK;
     }
 
+    // Only video track should be considered for Sync samples
+    if (track.mKind != Track::VIDEO) {
+        *sampleIndex = closestSampleIndex;
+        return OK;
+    }
+
     ssize_t prevSyncSampleIndex = closestSampleIndex;
     while (prevSyncSampleIndex >= 0) {
         const SampleInfo &info =
-- 
1.7.4.1

